UpSkill_App/UpSkill.Infrastructure/Data/Models/Coach.cs

```diff
@@ -1,9 +1,20 @@
 ﻿namespace UpSkill.Infrastructure.Data.Models
 {
-    using System.Linq;
+    using System;
+    using System.Collections.Generic;
 
     public class Coach : ApplicationUser
     {
-        public IQueryable<Course> Courses { get; set; }
+        public decimal PricePerSession { get; set; }
+
+        public ICollection<Course> Courses { get; set; }
```

Comment:
We need to initialize each ICollection in the Db entities. We can do it inline, not necessarily in the constructor:

public ICollection<LiveSession> LiveSessions { get; set; } = 
                                                   new HashSet<LiveSession>();

Yet, shouldn't all these be private readonly fields with IReadOnlyCollection-s to represent them outside? Plus behaviour to add to those private collections?

Reply:
Thank you for the feedback. It has been fixed. Now each ICollection property gets initialized in the constructor. Although I am still thinking about the readonly issue.